### PR TITLE
Simple template

### PR DESCRIPTION
**Pull request by tzufgoogle**
Changed the RVS template to use the OSM entities instead of the Wikidata it it currently using.

build, test and run works.

### CODE REVIEW

**Review by tzufgoogle — COMMENTED** [review]
ready

File: cabby/geo/walk.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def get_landmark_if_tag_exists(gdf: GeoDataFrame, tag: Text, main_tag:
     return None
 
 
-def pick_prominent_pivot(df_pivots: GeoDataFrame) -> Optional[GeoDataFrame]:
+def pick_prominent_pivot(df_pivots: GeoDataFrame) -> Optional[Dict]:
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:158]: Dict[???,???] (please prove key and value types)

File: cabby/geo/walk.py [hunk 2]
```diff
@@ -208,11 +209,9 @@ def get_pivot_near_goal(map: map_structure.Map, end_point: GeoDataFrame) -> \
     if prominent_poi is None:
         return None
 
-    return prominent_poi.to_dict('records')[0]
-
+    return prominent_poi
 
-def get_pivots(route: GeoDataFrame, map: map_structure.Map, end_point:
-               GeoDataFrame) -> Optional[Tuple[Dict, Dict]]:
+def get_pivot_along_route(route: GeoDataFrame, map: map_structure.Map) -> Optional[Dict]:
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:214]: Specify key and value types for Dict.

Comment (jasonbaldridge) [reply cabby/geo/walk.py:214]: As I look further (e.g. at get_pivots()), perhaps we need a pivot class that contains the relevant info? That would also help as code-as-documentation.

File: cabby/geo/walk.py [hunk 3]
```diff
@@ -245,13 +244,28 @@ def get_pivots(route: GeoDataFrame, map: map_structure.Map, end_point:
...
+    # Get pivot near the goal location.
+    main_pivot = get_pivot_along_route(route, map)
 
     # Get pivot near the goal location.
     near_pivot = get_pivot_near_goal(map, end_point)
     if main_pivot is None or near_pivot is None:
         return None
 
-    main_pivot = main_pivot.to_dict('records')[0]
+    main_pivot = main_pivot
```
Comment (jasonbaldridge) [inline cabby/geo/walk.py:268]: Remove redundant line.

File: cabby/rvs/generate_rvs.py [hunk 4]
```diff
@@ -58,45 +59,44 @@ def main(argv):
...
   print(f'Distance {path_distance} km | Bearing {bearing}')
   
-  # Get all Wikidata entities in Manhattan region and create dictionary from
-  # each entity's QID to the entity representation.
-  entities = {}
-  for result in query.get_geofenced_wikidata_items('Manhattan'):
-    entity = item.WikidataEntity.from_sparql_result(result)
-    entities[entity.qid] = entity
+  # Get all OSM in Manhattan region.
+  map = map_structure.Map("Manhattan", 18, "/mnt/hackney/data/cabby/poi/v1/")
```
Comment (jasonbaldridge) [inline cabby/rvs/generate_rvs.py:63]: Two things:
 - shouldn't the region be a command line option?
 - shouldn't reference specific on disk location

File: cabby/rvs/generate_rvs.py [hunk 5]
```diff
@@ -58,45 +59,44 @@ def main(argv):
...
-  pivot_qid = observe.get_pivot_poi(
-    start, target.location, list(entities.values()))
-  pivot = entities[pivot_qid]
-
-  start_pivot_bearing = util.get_bearing(start, pivot.location)
+  pois["distance"] = pois.centroid.apply(lambda x: util.get_distance_km(supplied_goal, x))
+
+  target = pois[pois['distance'].min()==pois['distance']].iloc[0]
+
+  print(f'\nChoosing entity {target["name"]} with OSM id {target["osmid"]} 'f'as the goal, which is {target["distance"]} from the supplied goal 'f'coordinates {supplied_goal}.\n')
```
Comment (jasonbaldridge) [inline cabby/rvs/generate_rvs.py:75]: Line >80 chars.

File: cabby/rvs/generate_rvs.py [hunk 6]
```diff
@@ -58,45 +59,44 @@ def main(argv):
...
+
+  # Get route. 
+  route = walk.compute_route(start, supplied_goal, map.nx_graph, map.nodes)
+
+  # Get the pivots.
+  result= walk.get_pivots(route, map, target)
+  if result is None:
+    print ("No pivots found.")
+    return
```
Comment (jasonbaldridge) [inline cabby/rvs/generate_rvs.py:84]: Since this is a main method, exit with the error instead of returning.

**Review by jasonbaldridge — CHANGES_REQUESTED** [review]
A few things to sort out.

File: cabby/geo/walk.py (hunk 3 above)
Comment (tzufgoogle) [reply cabby/geo/walk.py]: removed

File: cabby/rvs/generate_rvs.py (hunk 4 above)
Comment (tzufgoogle) [reply cabby/rvs/generate_rvs.py]: This is ofcourse not incorrect. I meant to changed it and forgot. 
 
Changed and added:
bazel-bin/cabby/rvs/generate_rvs \
  --start_lat 44.49582 --start_lon 11.33697 \
  --goal_lat 44.49268 --goal_lon 11.34365 \
	--region Bologna \
	--min_s2_level 18 \
	--directory "./cabby/geo/map_processing/poiTestData/" \

It runs and the result instruction makes sense.

File: cabby/rvs/generate_rvs.py (hunk 5 above)
Comment (tzufgoogle) [reply cabby/rvs/generate_rvs.py]: Changes to the way it was before.

File: cabby/rvs/generate_rvs.py (hunk 6 above)
Comment (tzufgoogle) [reply cabby/rvs/generate_rvs.py]: added:    
sys.exit("No pivots found.")

**Comment by tzufgoogle** [comment]
Build, test and run working

**Review by tzufgoogle — COMMENTED** [review]
Build, test and run working

File: cabby/geo/walk.py (hunk 2 above)
Comment (jasonbaldridge) [reply cabby/geo/walk.py:214]: Good for now, will consider ways to managed this better in future.

File: cabby/rvs/generate_rvs.py (hunk 4 above)
Comment (jasonbaldridge) [reply cabby/rvs/generate_rvs.py]: Great!

**Review by jasonbaldridge — APPROVED** [review]
LGTM

**Pull request merged**